### PR TITLE
Fix filelib:find_source for asn1 files

### PR DESCRIPTION
**Pull request by garazdawi**
This PR fixes `filelib:find_source/1` to work for asn1 source files and also fixes some minor issues (in public_key and eunit) where the source file for an Erlang/OTP module could not be found by `filelib:find_source/1`.

### CODE REVIEW

**Comment by KennethL** [comment]
I think is strange that filelib should have to know how the catalogues and makefiles for public_key is organized, which is definitely not according to any standard.

In the asn1 directory there are soure for asn.1 files but that does not say that the generated .erl files should land in the asn1 directory. Perhaps better to gave den under src or in src/asn1_generates

**Comment by garazdawi** [comment]
> In the asn1 directory there are soure for asn.1 files but that does not say that the generated .erl files should land in the asn1 directory. Perhaps better to gave den under src or in src/asn1_generates

Yes, this is exactly what this PR does.

The change in filelib is there for the .asn1 files that are directly compiled to .beam without any intermediary generated .erl files.

File: lib/public_key/asn1/Makefile [hunk 1]
```diff
@@ -18,6 +18,9 @@
 # %CopyrightEnd%
 #
 
+EGEN = ../src
```
Comment (u3s) [inline lib/public_key/asn1/Makefile:21]: is the EGEN needed for something? does not seem to be used in current Makefile...

File: lib/public_key/src/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+OTP-PUB-KEY.*
+PKCS-FRAME.*
```
Comment (u3s) [inline lib/public_key/src/.gitignore:2]: Maybe something related to asn1 folder content is not needed anymore in `./lib/public_key/.gitignore` and can be removed?

File: lib/public_key/asn1/Makefile (hunk 1 above)
Comment (garazdawi) [reply lib/public_key/asn1/Makefile:21]: it is used in `otp.mk` to know where to find/put generated erlang files.

File: lib/public_key/src/.gitignore (hunk 2 above)
Comment (garazdawi) [reply lib/public_key/src/.gitignore:2]: hmm, yes. will fix.

Comment (garazdawi) [reply lib/public_key/src/.gitignore]: I removed the new .gitignore and changed the rules in `lib/public_key/.gitignore` instead.

**Review by u3s — APPROVED** [review]

**Review by IngelaAndin — APPROVED** [review]

**Pull request merged**